internal/core/command/get.go

```diff
@@ -30,11 +31,21 @@ import (
 func NewGetCommand(
 	device contract.Device,
 	command contract.Command,
-	queryParams string,
 	context context.Context,
 	httpCaller internal.HttpCaller,
-	lc logger.LoggingClient) (Executor, error) {
-	urlResult := device.Service.Addressable.GetBaseURL() + strings.Replace(command.Get.Action.Path, DEVICEIDURLPARAM, device.Id, -1) + "?" + queryParams
+	lc logger.LoggingClient,
+	r *http.Request) (Executor, error) {
+
+	if r == nil {
+		return serviceCommand{}, errors.NewErrRequestParameterSupplied("")
```

Comment:
Adding to other comment about this error. I do not think this error fits this situation since it looks like query parameters are not mandatory anyway and if they are we should be able to specify which one(s) specifically are missing.